src/ccmain/control.cpp

```diff
@@ -721,9 +721,9 @@ void Tesseract::blamer_pass(PAGE_RES* page_res) {
         static_cast<IncorrectResultReason>(bl)),
         page_res->blame_reasons[bl]);
   }
-  if (page_res->misadaption_log.length() > 0) {
+  if (page_res->misadaption_log.size() > 0) {
```

Comment:
Maybe you can drop the `> 0` part.

Reply:
`.empty()` is better.

Reply:
Yes. In this case `! .empy()`, of course. `git grep "length().*0"` and `git grep "size().*0"` shows more code locations which could use `empty`.